### PR TITLE
[FIX] Self sent token transactions

### LINKED ISSUE #3780: iOS app incorrectly reports "Sent yourself WETH" when asset sent to external address
**Describe the bug**
I have worked with three different users who have been scammed. When assets are taken from their account, MetaMask on iOS reports the transfer as "Sent yourself WETH". I have also seen this for "Sent yourself USDC". In both cases, the user did NOT have the address for the recipient of these tokens.

This is really unfortunate because it gives them hope that their funds are just somewhere else they have access to.

**Screenshots**
Wallet history:

Only has one account in their account config:

The top transfer from above:
https://etherscan.io/tx/0xd51dfc97269b33cfbe372aa73f4741e72fba72f32050b69b5b48686d5b7f9039

**To Reproduce**
I am not sure how to reproduce, but the send in question was not originated by MetaMask on iOS, so it might be confused about transactions that were originated elsewhere?

**Expected behavior**
Account history should reflect that the ETH was sent to a third party and not "Yourself"

**Smartphone (please complete the following information):**
 - Device: iPhone
 - App Version: v4.0.1 (814)

-------------------------------------------------------------

_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

Send token transaction is being labelled as "Sent Yourself X" instead of "Sent X" when an user sends tokens to another address instead of the same address as the sender address.

Cases to confirm that the transaction is being labelled correctly:

-  Sent ETH
- Sent token (WETH, UNI, MKR)
- Sent yourself ETH
- Sent yourself token (WETH, UNI, MKR)
- Sent a collectible
- Sent yourself a collectible
- Contract deploy
- Swap transaction
- Smart contract interaction

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #3780

### CODE REVIEW

File: app/util/transactions/index.js [hunk 1]
```diff
@@ -315,27 +315,28 @@ export async function getTransactionActionKey(transaction, chainId) {
  * @returns {string} - Transaction type message
  */
 export async function getActionKey(tx, selectedAddress, ticker, chainId) {
-	if (tx && tx.isTransfer) {
-		const selfSent = safeToChecksumAddress(tx.transaction.from) === selectedAddress;
-		const translationKey = selfSent ? 'transactions.self_sent_unit' : 'transactions.received_unit';
+	let tokenUnit = ticker;
```
Comment (andreahaku) [inline app/util/transactions/index.js:318]: why don't you move lines 318-325 inside the `if` at line 327 as they are only used there. Am I wrong?

Comment (gantunesr) [reply app/util/transactions/index.js:318]: I think you're correct good sir, it makes no sense to have it outside the `if`

Comment (gantunesr) [reply app/util/transactions/index.js]: commit: `14403a112f60538e66c27aca28b67e6104a1fa32`

**Review by andreahaku — APPROVED** [review]

File: app/util/transactions/index.js (hunk 1 above)
Comment (andreahaku) [reply app/util/transactions/index.js]: thanks :)

**Comment by epheph** [comment]
Bump? We're so close!

**Comment by epheph** [comment]
Bump

**Review by cortisiko — APPROVED** [review]
🌮 🙇

**Pull request merged**